src/lib.rs

```diff
@@ -285,179 +302,31 @@ fn run_commands<C: Command>(runner: &Runner<C>) -> CommandHandle {
-                    name: command_name.clone(),
-                    message: OutputMessagePayload::Done(status.code()),
-                });
-
-                match options.restart {
-                    RestartOptions::Continue => {
-                        return (command_name, Some(status));
-                    }
-                    RestartOptions::Restart => {
-                        if status.success() {
-                            return (command_name, Some(status));
-                        }
-                    }
-                    RestartOptions::Kill => {
-                        if !status.success() {
-                            let _ = kill_trigger.initiate_kill();
-                        }
-                        return (command_name, Some(status));
-                    }
-                };
-            }
-            Err(e) => {
-                let _ = send_chan.send(OutputMessage {
-                    name: command_name.clone(),
-                    message: OutputMessagePayload::Error(e),
-                });
-                return (command_name, None);
-            }
-        }
-    })
+fn check_command(exec_name: &str) -> Result<(), CommandError> {
+    which::which(exec_name).map_err(|_| CommandError::CommandNotFound(exec_name.to_string()))?;
+    Ok(())
 }
 
-fn kill_thread(kill_trigger: &kill_barrier::KillBarrier, child: Arc<Mutex<process::Child>>) {
-    let _ = kill_trigger.wait();
+#[cfg(test)]
+mod test {
+    use crate::Command;
 
-    let lock_res = child.lock();
-    if let Ok(mut locked_child) = lock_res {
-        let _ = locked_child.kill();
-    }
-}
+    #[test]
+    fn command_not_found() {
+        let cmd = super::ConsoleCommand::from_string("test", "bogus_cmd_not_found");
 
-fn read_stream<R>(
-    cmd_name: &str,
-    send_chan: mpsc::Sender<OutputMessage>,
-    reader: &mut R,
-    is_stdout: bool,
-) where
-    R: BufRead,
-{
-    loop {
-        let line = line_parse::get_line(reader);
-        match line {
-            Ok(Some(line_vec)) => {
-                let _ = send_chan.send(OutputMessage {
-                    name: cmd_name.to_string(),
-                    message: if is_stdout {
-                        OutputMessagePayload::Stdout(line_vec.0, line_vec.1)
-                    } else {
-                        OutputMessagePayload::Stderr(line_vec.0, line_vec.1)
-                    },
-                });
-            }
-            Ok(None) => {
-                return;
-            }
-            Err(e) => {
-                let _ = send_chan.send(OutputMessage {
-                    name: cmd_name.to_string(),
-                    message: OutputMessagePayload::Error(e),
-                });
+        match cmd {
+            Err(super::CommandError::CommandNotFound(name)) => {
+                assert_eq!(
+                    &name, "bogus_cmd_not_found",
+                    "Command Not Found Error has wrong command name"
```

Comment:
Just an opinion, thanks to Rust's correct line numbers in error messages I have found it unnecessary to add such descriptions to assertions. The only exception is when asserting in a loop and you want to know for which input it failed.